examples/src/demo_client.rs

```diff
@@ -277,19 +279,44 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     println!("\t{}\n", read_causal_response);
 
-    block_proposal_collection_ids.extend(read_causal_response.collection_ids);
+    block_validation_collection_ids.extend(read_causal_response.collection_ids);
+
+    println!("\tFound {} collections from read causal which will be combined with the {} extra collections from the block proposal\n", block_validation_collection_ids.len(), extra_collections.len());
+
+    block_validation_collection_ids.extend(extra_collections);
+
+    println!("\tProposed block included the following collections before compressing proposal:\n");
+    let mut result = "\t\t*** Block Proposal Collections ***".to_string();
+    for id in block_proposal_collection_ids.clone() {
+        result = format!("{}\n\t\t|-id=\"{}\"", result, id);
+    }
+    println!("{}", result);
+
+    println!(
+        "\n\tBlock validation found the following collections after decompressing proposal:\n"
+    );
+    let mut result = "\t\t*** Block Validation Collections ***".to_string();
+    for id in block_validation_collection_ids.clone() {
+        result = format!("{}\n\t\t|-id=\"{}\"", result, id);
+    }
+    println!("{}", result);
 
-    println!("\tFound {} collections from read causal which will be combined with the {} extra collections from the block proposal", block_proposal_collection_ids.len(), extra_collections.len());
-    block_proposal_collection_ids.extend(extra_collections);
+    if block_proposal_collection_ids == block_validation_collection_ids {
+        println!("\n\tThey match in value and order! Moving on to find the transactions...\n");
+    } else {
+        println!("\n\tThey dont match! Aborting...\n");
+        return Ok(());
+    }
```

Comment:
I would add a code comment at the top of this that we're cheating: we're comparing `block_validation_collection_ids` which is a validator determined artefact, and `block_proposal_collection_ids` which is a proposer artefact. In production, the consensus would not have access to that second artefact at any node but the proposer, and we only do show this here for didactic purposes.